src/lib/mdns/minimal/tests/TestQueryReplyFilter.cpp

```diff
@@ -0,0 +1,132 @@
+const QNamePart kName2[] = { "something", "else" };
+const QNamePart kName3[] = { "more", "specific", "some", "local" };
+const QNamePart kName4[] = { "this", "is", "a", "test" };
+
+template <size_t N>
+QueryData buildQueryData(QType qType, QClass qClass, const uint8_t (&query)[N])
+{
+    return QueryData(qType, qClass, false, query, BytesRange(query, query + N));
+}
+
+void TestQueryReplyFilter(nlTestSuite * inSuite, void * inContext)
+{
+    const uint8_t query[] = {
+        4, 's', 'o', 'm', 'e',      //
+        5, 'l', 'o', 'c', 'a', 'l', //
+        0                           //
+    };
+
+    // sanity test that the serialized qname was build correctly
+    NL_TEST_ASSERT(inSuite, SerializedQNameIterator(BytesRange(query, query + sizeof(query)), query) == FullQName(kName1));
+    NL_TEST_ASSERT(inSuite, SerializedQNameIterator(BytesRange(query, query + sizeof(query)), query) != FullQName(kName2));
+    NL_TEST_ASSERT(inSuite, SerializedQNameIterator(BytesRange(query, query + sizeof(query)), query) != FullQName(kName3));
+    NL_TEST_ASSERT(inSuite, SerializedQNameIterator(BytesRange(query, query + sizeof(query)), query) != FullQName(kName4));
+
+    // Acceptable cases
+    NL_TEST_ASSERT(
+        inSuite,
+        QueryReplyFilter(buildQueryData(QType::ANY, QClass::ANY, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName1)));
+    NL_TEST_ASSERT(
+        inSuite,
+        QueryReplyFilter(buildQueryData(QType::A, QClass::ANY, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName1)));
+
+    NL_TEST_ASSERT(
+        inSuite,
+        QueryReplyFilter(buildQueryData(QType::ANY, QClass::IN, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName1)));
+
+    NL_TEST_ASSERT(
+        inSuite, QueryReplyFilter(buildQueryData(QType::A, QClass::IN, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName1)));
+
+    // Reject cases
+    NL_TEST_ASSERT(
+        inSuite,
+        !QueryReplyFilter(buildQueryData(QType::ANY, QClass::ANY, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName2)));
+
+    NL_TEST_ASSERT(
+        inSuite,
+        !QueryReplyFilter(buildQueryData(QType::ANY, QClass::ANY, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName3)));
+
+    NL_TEST_ASSERT(
+        inSuite,
+        !QueryReplyFilter(buildQueryData(QType::ANY, QClass::ANY, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName4)));
+
+    NL_TEST_ASSERT(
+        inSuite,
+        !QueryReplyFilter(buildQueryData(QType::AAAA, QClass::ANY, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName1)));
+
+    NL_TEST_ASSERT(
+        inSuite,
+        !QueryReplyFilter(buildQueryData(QType::SRV, QClass::IN, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName1)));
+
+    NL_TEST_ASSERT(
+        inSuite,
+        !QueryReplyFilter(buildQueryData(QType::PTR, QClass::ANY, query)).SendAnswer(QType::A, QClass::IN, FullQName(kName1)));
+}
+
+void TestLongerQueryPath(nlTestSuite * inSuite, void * inContext)
```

Comment:
Suggest a test case for a Qname with mixed case as well.

Reply:
What is the intent?

Does the spec require case insensitive or is this to check that we are indeed case sensitive (my code is right now)?